### DISCUSSION TITLE
How to use FileAttachment on a gif file in root directory?

### DISCUSSION THREAD

**Discussion by gganapat** [observablehq/framework#1077] [Q&A] [answered]
Hi:

I was able to run a wolframscript script in a data loader which along with sending data to stdout also generates a plot in a separate following command. But the plot is stored in the root directory. Is there a way to load an image file from the root directory?

Thx!

**Comment by gganapat** [comment]
I was able to set the directory in wolframscript to store the plot in /docs/data. So, the data loader was able to pick it up. Great!

**Reply by Fil** [reply]
I'm curious to see how the data loader looks like. Mathematica can be incredibly useful

**Reply by gganapat** [reply]
Will do once I get back home later today.

G-

**Comment by gganapat** [comment, marked as answer]
The data loader file for wolframscript (assuming it is your $PATH) is very simple -- you have to name it ending in ```.exe``` (even if you are on a Mac like I am!), so my file was named ```wolfram.txt.exe``` and stored in docs/data folder. The resulting plot is also stored in the same folder by setting the right path in Mathematica. The resulting outputs can be simply referenced in markdown and html tag. Very convenient - no need for display(xxx) in js block, etc.

```
#!/usr/bin/env wolframscript
b = Integrate[1/(x^4 + 1), x]/.{x->2.0}
Print[b]
cwd=Directory[]
dir = StringJoin[cwd<>"/docs"<>"/data"]
Export[StringJoin[dir<>"/test.gif"], Plot[Sin[x], {x, 0, 10}]]
```

The wolfram.md file is

<img width="656" alt="Screenshot 2024-03-16 at 12 42 06 PM" src="https://github.com/observablehq/framework/assets/46061218/549b9c98-5171-4a59-ab4a-2ad6da531525">

And the result is 
<img width="629" alt="Screenshot 2024-03-16 at 12 25 52 PM" src="https://github.com/observablehq/framework/assets/46061218/86f2fc17-5d35-4013-9753-73a8e70a303c">

**Reply by Fil** [reply]
Very cool, thanks for sharing!

We used ".exe" for executable data loaders because it evokes "executable", but it's true that it can create a bit of confusion since that extension is also associated to Windows-executable files. In the case of Observable Framework, the only rule for a .exe data loader is that "it must run on the machine".

**Comment by gganapat** [comment]
That's fantastic! I didn't realize you are one of the contributors! I think data loaders is a game changer and will definitely encourage more people to use Observable Framework in place of say, Svelte which is also a great product but it doesn't have this feature which is very distinct.

gani-
